runtime/minor_gc.c

```diff
@@ -518,13 +519,97 @@ void caml_empty_minor_heap_domain_clear (struct domain* domain, void* unused)
   clear_table ((struct generic_table *)&minor_tables->ephe_ref);
   clear_table ((struct generic_table *)&minor_tables->custom);
 
+  return;
+}
+
+void caml_adjust_global_minor_heap(int participating_domains) {
+  // Calculate the size of the existing mapping
+  int global_minor_heap_size = caml_global_minor_heap_limit - caml_global_minor_heap_start;
+  // Now using the number of participating domains, we calculate the new size
+  int new_global_minor_heap_size = participating_domains*Bsize_wsize(global_minor_heap_wsz_per_domain) + caml_global_minor_heap_start;
```

Comment:
I think adding `caml_global_minor_heap_start` wasn't meant here
It just works because everything is page aligned (and thus, is effectively adding zero the the number).